aten/src/ATen/native/ScatterGatherChecks.h

```diff
@@ -31,42 +31,31 @@ static void scatter_gather_dtype_check(
 }
 
 // Used for `gather`-like methods
+// self here means the input tensor
 // Test:
-// 1. index.size(d) == self.size(d) for all d != dim
-// 2. index.size(d) <= src.size(d) for all d != dim
-// 3. index.dim() == self.dim() == src.dim()
+// 1. index.size(d) <= self.size(d) for all d != dim
+// 2. index.dim() == self.dim()
 static C10_UNUSED void gather_shape_check(const Tensor& self, int64_t dim,
-  const Tensor& index, const Tensor& src
+  const Tensor& index
```

Comment:
Post offline discussion with @ysiraichi, it was a little confusing because `scatter` has an `src` as an input tensor, while `gather` doesn't have an `src` tensor. We moved the checks for the `out` tensor to the `meta` func of `gather`.